include/Jit/LLILCJit.h

```diff
@@ -174,6 +180,32 @@ class NullResolver : public llvm::RuntimeDyld::SymbolResolver {
   }
 };
 
+class ObjectLoadListener {
```

Comment:
FWIW, that would be fine with me -- so long as the template declaration and definition are in the same file, there's no worry about referencing an undefined instantiation.